src/widgets/modalChildren/PatientEditModal.js

```diff
@@ -23,11 +26,9 @@ export const PatientEditModalComponent = ({
   onUpdateForm,
   nameNoteIsValid,
 }) => {
-  const onSaveWithForm = completedForm => {
-    const id = generateUUID();
-
-    onSave({ id, ...completedForm });
-    onSaveSurvey(id);
+  const onSaveWithForm = () => {
+    onSaveForm();
+    onSaveSurvey();
```

Comment:
With this, creating a new name is not creating a name note:

I added this because when creating a new patient, we don't have the ID of the name yet (we create the name from the `completedForm` of the `FormControl` and we create the name note from the `formData` of the `JSONForm`.. but we need an `id` for the patient before creating the `name_note`, so I added an optional `id` to these for this case.

We have no `id` to pass into `createSurveyNameNote` when opening, so we don't have the `id` of the `name` when we call `onSaveSurvey`.

Think we need to add a prop `isCreating`, and if we're creating, do the same as before- generate a UUID and pass that into the actions, otherwise, don't pass it in

Reply:
good point - I've updated, so when we extract patient from `ownProps`, if there isn't one then generate a new ID and use that throughout. No need for `isCreating` unless you think it would be better to be explicit?

5f45636b